ui/panels-plugin/src/components/OptionsEditorTabs/OptionsEditorTabs.tsx

```diff
@@ -65,17 +70,27 @@ export const OptionsEditorTabs = ({ tabs }: OptionsEditorTabsProps) => {
   // Normalize the common tabs that are managed via constants in this file
   // and custom tabs that bring their own config into a consistent shape for
   // rendering.
-  const commonTabs = TAB_CONFIG.filter((tabConfig) => {
-    // Only include common tabs that are specified.
-    return !!tabs[tabConfig.id];
-  }).map((tabConfig) => {
-    return {
-      ...tabConfig,
-      ...tabs[tabConfig.id],
-    };
-  });
-  const otherTabs = tabs?.other || [];
-  const normalizedTabs = [...commonTabs, ...otherTabs];
+  const normalizedTabs = TAB_CONFIG.reduce((combinedTabs, tabConfig) => {
```

Comment:
This `reduce` is a little harder to read, but needed because we're now intermingling the "common" and "custom" tabs, with the order being: 
- visual common tabs
- custom tabs
- json common tab